java/arcs/core/analysis/RecipeGraphFixpointIterator.kt

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.analysis
+
+import arcs.core.data.HandleConnectionSpec
+import arcs.core.data.Recipe
+
+/** An abstract class that implements dataflow analysis over abstract values of type [V]. */
+abstract class RecipeGraphFixpointIterator<V : AbstractValue<V>>(val bottom: V) {
+    /** Results of the fixpoint computation. */
+    class FixpointResult<V : AbstractValue<V>>(
+        private val bottom: V,
+        private val nodeValues: Map<RecipeGraph.Node, V>
+    ) {
+        /** Returns the value for the given particle.  */
+        fun getValue(particle: Recipe.Particle): V =
+            nodeValues[RecipeGraph.Node.Particle(particle)] ?: bottom
+
+        /** Returns the value for the given handle. Returns null if the value is BOTTOM. */
```

Comment:
Nit: backticks around BOTTOM

Reply:
this comment was slate. Remove the second line.